hw3/3.1/main.c

```diff
@@ -0,0 +1,131 @@
+#include <locale.h>
+#include <stdbool.h>
+
+#define TEST_ARRAYS_NUMBER 10
+#define ARRAYS_SIZE 50
+
+// sorts segment [leftBound, rightBound] in array
+void insertionSort(int *array, const int leftBound, const int rightBound) {
+    for (int i = leftBound + 1; i <= rightBound; i++) {
+        for (int j = i - 1; j >= leftBound && array[j] > array[j + 1]; j--) {
+            const int tmp = array[j];
+            array[j] = array[j + 1];
+            array[j + 1] = tmp;
+        }
+    }
+}
+
+// parts segment [leftBound, rightBound] of array
+// returns right bound of left part and left bound of right part
+void partition(int *array, const int leftBound, const int rightBound,
+               int *pLeftIndex, int *pRightIndex, const int pivot) {
+
+    int leftIndex = leftBound;
+    int rightIndex = rightBound;
+    while (leftIndex <= rightIndex) {
+        while (leftIndex <= rightBound && array[leftIndex] < pivot) {
+            leftIndex++;
+        }
+        while (rightIndex >= leftBound && array[rightIndex] >= pivot) {
+            rightIndex--;
+        }
+        if (leftIndex <= rightIndex) {
```

Comment:
```suggestion
        if (leftIndex < rightIndex) {
```
Зачем в случае равенства индексов перестановку проводить?